src/Core/SettingsChangesHistory.cpp

```diff
@@ -81,6 +81,7 @@ const VersionToSettingsChangesMap & getSettingsChangesHistory()
             {"use_skip_indexes_if_final", 0, 1, "Change in default value of setting"},
             {"use_skip_indexes_if_final_exact_mode", 0, 1, "Change in default value of setting"},
             {"min_outstreams_per_resize_after_split", 0, 24, "New setting."},
+            {"count_matches_stop_at_empty_match", false, false, "New setting."},
```

Comment:
See l. 66-67. This needs to be

```cpp
{"count_matches_stop_at_empty_match", true, false, "New setting."},
```

so that users with [the compatibility setting](https://clickhouse.com/docs/operations/settings/settings#compatibility) configured to something pre-25.6 will get the previous behavior.